Added BasicRows, fix py3 StringIO issue

ilvalle opened the pull request:
Added as_dict() as_csv() as_json() as_xml() as_list() to IterRows

gi0baro commented:
I would ask why you need `abc` and `ABCMeta` for this?

Beside the StringIO fix, which I like,  there are quite a lot of stuffs you wrote that are not python3 compatible, please check out https://github.com/web2py/pydal/wiki/Write-py2py3-compatible-code

ilvalle commented:
Actually, I've just refactored Rows by introducing the super class BasicRows, and I added a few tests, many methods weren't tested at all (this is the reason because I spotted the StringIO issue)
I've introduced abc to explicitly mark the class as Abstract, now I removed it since it isn't compatible with py3

gi0baro commented:
Actually `abc` is python3 compatible, the objection was that is pointless use it without implements the decorators inside the class. Anyway, I will merge this and then update the code with python3 corrections.
